src/compositor/windowpixmapitem.cpp

```diff
@@ -556,6 +556,7 @@ QSGNode *WindowPixmapItem::updatePaintNode(QSGNode *oldNode, UpdatePaintNodeData
             m_textureProvider = prov;
 
             prov->fbo = 0;
+            prov->t = 0;
```

Comment:
AFAICT SnapshotTextureProvider destructor assumes that those pointers are valid and can be deleted.
Wouldn't it be better to clear the pointers in contructor instead of scattering set-to-null code here and there?